Modules/_sqlite/connection.c

```diff
@@ -2005,10 +2005,26 @@ pysqlite_connection_iterdump_impl(pysqlite_Connection *self,
     }
     PyObject *args[2] = {(PyObject *)self, filter};
     PyObject *kwnames = PyTuple_New(1);
-    PyTuple_SET_ITEM(kwnames, 0, PyUnicode_FromString("filter"));
+    PyObject *py_filter = NULL;
+
+    if (!kwnames) {
+        goto error;
+    }
+    py_filter = PyUnicode_FromString("filter");
+    if (!py_filter) {
+        goto error;
+    }
+    PyTuple_SET_ITEM(kwnames, 0, py_filter);
+
     PyObject *retval = PyObject_Vectorcall(iterdump, args, 1, kwnames);
     Py_DECREF(iterdump);
     return retval;
+
+error:
+    Py_DECREF(iterdump);
+    Py_DECREF(args);
```

Comment:
Args is an array of PyObject's, so we cannot directly decref it :) We don't need to worry about the refcount for the incoming PyObject's to `pysqlite_connection_iterdump_impl()` (that is: `self` and `filter`).

```suggestion
```